lib/models/repository.js

```diff
@@ -140,8 +140,16 @@ export default class Repository {
     return this.workingDirectoryPath;
   }
 
+  setGitDirectoryPath(gitDirectoryPath) {
+    this._gitDirectoryPath = gitDirectoryPath;
+  }
+
   getGitDirectoryPath() {
-    return path.join(this.getWorkingDirectoryPath(), '.git');
+    if (this._gitDirectoryPath) {
+      return this._gitDirectoryPath;
+    } else {
+      return path.join(this.getWorkingDirectoryPath(), '.git');
```

Comment:
I left this mostly for tests in case they didn't go through the normal setup process — but perhaps they all do? I didn't try it.